homeassistant/components/amcrest/camera.py

```diff
@@ -89,18 +90,27 @@
     }
 )
 
+
+class CameraEntry(NamedTuple):
```

Comment:
It is not really a competition of which is used more. I think NamedTuple is a good fit here.

Remark entityDescription is not a good fit, since it contains other data.

Reply:
That is fair, I tend to prefer dataclasses, but the typed named tuple is just as good!

Reply:
Each have their strengths. Dataclass uses more memory but is not read-only (except if you use frozen=true) like the tuple. I (we) try to use tuples for read-only items to ensure they are not changed by accident.